File: EventService/EventService/Models/Event.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;

namespace EventService.Models
{
    public class Event
    {
        public int Id { get; set; }
        public string Type { get; set; }
        public string Venue { get; set; }
        public DateTime Date { get; set; }
        [NotMapped]
        public ICollection<int> Headliners { get; set; }
        [NotMapped]
        public ICollection<int> Supporters { get; set; }
    }
}

File: musicsocialnetwork-client/src/models/Event.ts
export interface EventData {
  id: number;
  date: Date;
  venue: string;
  headliners: Array<number>;
  supporters: Array<number>;
}

export const defaultEvent: EventData = {
  id: 0,
  date: new Date(),
  venue: '',
  headliners: [],
  supporters: []
};

export interface UserEvent {
  eventId: number;
  markEventType: number;
  venue: string;
  date: Date;
}